### PR TITLE
[Join] Add testcases for join @open sesame 11/26 08:44

### PR DESCRIPTION
**Pull request by gichan-jang**
Add testcases for join element.
Line coverage: 92.4%

Signed-off-by: gichan-jang <gichan2.jang@samsung.com>

Self evaluation:
1. Build test: [ *]Passed [ ]Failed [ ]Skipped
2. Run test: [ *]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: tests/gstreamer_join/unittest_join.cc [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+      ("appsrc name=appsrc_0 ! other/tensor,dimension=(string)3:4:2:2,type=(string)int32,framerate=(fraction)0/1 ! join.sink_0 "
+       "appsrc name=appsrc_1 ! other/tensor,dimension=(string)3:4:2:2,type=(string)int32,framerate=(fraction)0/1 ! join.sink_1 "
+       "join name=join ! other/tensor,dimension=(string)3:4:2:2, type=(string)int32, framerate=(fraction)0/1 ! "
+       "tensor_sink name=sinkx async=false");
+
+  GstElement *pipeline = gst_parse_launch (str_pipeline, NULL);
+  EXPECT_NE (pipeline, nullptr);
+
+  join_handle = gst_bin_get_by_name (GST_BIN (pipeline), "join");
+  EXPECT_NE (join_handle, nullptr);
```
Comment (dongju-chae) [inline tests/gstreamer_join/unittest_join.cc:181]: When considering this unittest's purpose, I think the checking for `pipeline` & `join_handle` should be `assert`, not `expect`

File: tests/gstreamer_join/unittest_join.cc [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
```
Comment (dongju-chae) [inline tests/gstreamer_join/unittest_join.cc:209]: ditto

**Review by myungjoo — CHANGES_REQUESTED** [review]
Need more test cases:
1. GSTCAPs mismatched (negative case)
2. Unspecified GSTCAP that is determined by JOIN (positive)
3. Unspecified GSTCAP that is determined by JOIN and mismatched (negative case)

**Review by helloahn — APPROVED** [review]
LGTM

File: tests/gstreamer_join/unittest_join.cc (hunk 1 above)
Comment (gichan-jang) [reply tests/gstreamer_join/unittest_join.cc]: Thanks. Updated.

**Comment by gichan-jang** [comment]
> Need more test cases:
> 
> 1. GSTCAPs mismatched (negative case)
> 2. Unspecified GSTCAP that is determined by JOIN (positive)
> 3. Unspecified GSTCAP that is determined by JOIN and mismatched (negative case)

Suggested TCs are added in tests/gstreamer_join/runTest.sh

**Review by myungjoo — APPROVED** [review]
LGTM

**Pull request merged**